FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -2261,6 +2261,15 @@ - (nullable FIRUser *)getStoredUserForAccessGroup:(NSString *_Nullable)accessGro
 #endif  // TARGET_OS_WATCH
     user = [unarchiver decodeObjectOfClass:[FIRUser class] forKey:userKey];
   } else {
+#if TARGET_OS_TV
+    if (self.shareAuthStateAcrossDevices) {
+      FIRLogError(kFIRLoggerAuth, @"I-AUT000001",
```

Comment:
Is this better than making the API unavailable for tvOS? Why not give customers a compile time error instead of a runtime error? cc: @ryanwilson

Reply:
I realized I should verify if tvOS keychain sharing works on between tvOS apps on the same device. It works so this Auth API should work on tvOS when `shareAuthStateAcrossDevices` is false. Because of this, I don't think we should mark the API unavailable because we are blocking a valid use case on tvOS (sharing a user between tvOS apps) .

Reply:
Thanks for the clarification. Please make that clear in the note and log message.

Reply:
Reworked in latest commits.